tldw_Server_API/app/core/AuthNZ/llm_budget_middleware.py

```
@@ -47,27 +61,78 @@ async def dispatch(self, request: Request, call_next: Callable) -> Response:
+                            salt=key,
+                            iterations=100_000,
+                        ).hex()
+                        if digest not in digests:
+                            digests.append(digest)
+                    if digests:
+                        pool = await get_db_pool()
+                        placeholders = ",".join("?" for _ in digests)
+                        query = (
+                            f"SELECT id, user_id FROM api_keys "
+                            f"WHERE key_hash IN ({placeholders}) AND status = ? "
+                            f"ORDER BY created_at DESC LIMIT 1"
```

Ruff (0.14.2)

Possible SQL injection vector through string-based query construction

(S608)